src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -37,6 +37,16 @@
 extern XmaSingleton *g_xma_singleton;
 
 namespace xma_core {
+    const std::map<XmaSessionType, const std::string> sessionMap = {
```

Comment:
The sessionMap value should be static and access to it should be done via a function.   Code support can be extremely difficult if every the signature for the map should change.

For pretty print functionality like this, the following code might be of interest:

```
std::string get_pretty_name(XmaSessionType eSessionType)
{
  std::map<XmaSessionType, const std::string>::const_iterator it = sessionMap.find(eSessionType);

  if (it == sessionMap.end()) {
    return std::string("Unknown XmaSessionType");
  }

  return it->second;
}
```

Reply:
excellent suggestion. done